bin/decomm.rb

```diff
@@ -0,0 +1,172 @@
+# no longer exists.
+#
+# Requires the following Rubygems (`gem install $GEM`):
+#   - ridley
+#   - flapjack-diner
+#
+# Requires a Sensu configuration snippet:
+# Note: Sensu user has to be an owner of the file '/path/to/spocks/key.pem'
+#       you can simply copy a key file to '/etc/sensu/ssl' and set appropriate ownership and permissions
+#
+#       chown sensu /etc/sensu/ssl/key.pem
+#       chmod 400 /etc/sensu/ssl/key.pem
+#
+#   {
+#     "decomm": {
+#       "chef": {
+#         "in_use": true,
+#         "server_url": "https://api.opscode.com:443/organizations/foobar",
+#         "client_name": "foo.example.com",
+#         "client_key": "/etc/sensu/ssl/key.pem"
+#       },
+#       "flapjack": {
+#         "in_use": true,
+#         "server_url": "127.0.0.1:3081"
+#       }
+#     }
+#   }
+#
+# Best to use this handler with a filter:
+# Note: Only keepalive events with occurrences value greater than 10 will be handled by this handler
+#       10 * 30s = 300s or 5min (30s represents keepalive check interval)
+#
+#   {
+#     "filters": {
+#       "keepalives": {
+#         "attributes": {
+#           "check": {
+#             "name": "keepalive"
+#           },
+#           "occurrences": "eval: value > 10"
+#         }
+#       }
+#     },
+#     "handlers": {
+#       "decomm": {
+#         "type": "pipe",
+#         "command": "decomm.rb",
+#         "filter": "keepalives",
+#         "timeout": 90
+#       }
+#     }
+#   }
+#
+# Add decomm handler in the client configuration (e.g. client.rb)
+#   {
+#     "client": {
+#       "name": "bar.example.com",
+#       "address": "1.1.1.1",
+#       "subscriptions": [
+#         "base"
+#       ],
+#       "keepalive": {
+#         "type": "metric",
+#         "thresholds": {
+#           "warning": 120,
+#           "critical": 180
+#         },
+#         "handlers": [
+#           "flapjack",
+#           "decomm"
+#         ],
+#         "refresh": 60
+#       }
+#     }
+#   }
+#
+# Copyright 2013 Heavy Water Operations, LLC.
+#
+# Released under the same terms as Sensu (the MIT license); see
+# LICENSE for details.
+
+require 'sensu-handler'
+require 'ridley'
+require 'flapjack-diner'
+
+class Decomm < Sensu::Handler
+  def handler_name
+    File.basename($PROGRAM_NAME, File.extname($PROGRAM_NAME))
+  end
+
+  def chef_node_exists?
+    retried = 0
+    begin
+      Ridley::Logging.logger.level = Logger.const_get 'ERROR'
+
+      Ridley.open(
+        server_url: settings[handler_name]['chef']['server_url'],
+        client_name: settings[handler_name]['chef']['client_name'],
+        client_key: settings[handler_name]['chef']['client_key'],
+        ssl: {
+          verify: settings[handler_name]['chef']['client_key'].nil? ? true : settings[handler_name]['chef']['verify_ssl']
+        }
+      ) do |r|
+        r.node.find(@event['client']['name']) ? true : false
+      end
+    # FIXME: Why is this necessary?  Ridley works fine outside of Sensu
+    rescue Celluloid::Error
```

Comment:
Pretty sure if you move `Celluloid.boot` to before it works without rescue, (it does for the other checks that now use ridley).